index/scorch/persister.go

```diff
@@ -39,15 +39,15 @@ type notificationChan chan struct{}
 func (s *Scorch) persisterLoop() {
 	defer s.asyncTasks.Done()
 
-	var notifyChs []notificationChan
+	var persistWatchers []*epochWatcher
 	var lastPersistedEpoch uint64
 OUTER:
 	for {
 		select {
 		case <-s.closeCh:
 			break OUTER
 		case notifyCh := <-s.persisterNotifier:
-			notifyChs = append(notifyChs, notifyCh)
+			persistWatchers = append(persistWatchers, notifyCh)
```

Comment:
i'm a bit confused as to why here we unconditionally append to persistWatchers, but down at the end we apply logic to the same operation

Reply:
yea..we should ideally handle it the same way here too..